graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/response/ResponseSchemaEntry.java

```diff
@@ -16,9 +16,44 @@
  */
 package org.graylog.plugins.views.search.rest.scriptingapi.response;
 
+import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public record ResponseSchemaEntry(@JsonProperty String name,
-                                  @JsonProperty String type,
-                                  @JsonProperty String field) {
+import java.util.Locale;
+import java.util.Optional;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public record ResponseSchemaEntry(
```

Comment:
I've never done it for records, but I would make constructor private if possible so that only static methods can be used to construct instances.

Reply:
I don't think it's possible. We would have to convert the class to an normal/autovalue class and get rid of the record structure.